hal/src/nRF52840/ble_hal.cpp

```diff
@@ -2545,6 +2536,15 @@ int BleObject::ConnectionsManager::processSecurityEventFromThread(const ble_evt_
         linkEvent.params.pairing_status.bonded = authStatus.bonded;
         linkEvent.params.pairing_status.lesc = authStatus.lesc;
         notifyLinkEvent(linkEvent);
+        if (authStatus.lesc) {
+            /* To protect a device's private key, a device should implement a method to
+            * prevent an attacker from retrieving useful information about the device's private
+            *  key. For this purpose, a device should change its private key after every pairing
+            * (successful or failed). Otherwise, it should change its private key whenever S +
+            * 3F > 8, where S is the number of successful pairings and F the number of
+            * failed attempts since the key was last changed. */
+            generateEcdhKeyPair();
```

Comment:
This is not going to work because `ecdhContextInit_` is not reset anywhere. We should probably just remove the `ecdhContextInit_` state variable and generate new keys every time in `publicKeysPrepare()` perhaps?

Reply:
`ecdhContextInit_` is reset once enter the `generateEcdhKeyPair()` and set if new ECDH key pair is generated successfully. `ecdhContextInit_` is used elsewhere to check the ECDH key pair validity in prior to invoking the key, such as computing the DHKey.